src/main/java/fr/insee/pogues/transforms/visualize/PoguesJSONToPoguesJSONDerefImpl.java

```diff
@@ -118,55 +141,91 @@ private String questionnaireJavaToString(Questionnaire quest) {
+        for (ComponentType seq : questionnaire.getChild()) {
+            if (seq.getId().equals(reference)) {
                 break;
             }
             indexOfModification++;
-        };
-        logger.info("Index to modify", indexOfModification);
+        }
+        logger.info("Index to modify {}", indexOfModification);
         // Suppression of the questionnaire reference
-        quest.getChild().remove(indexOfModification);
+        questionnaire.getChild().remove(indexOfModification);
         // Insertion of the sequences
         for (int i=0; i<refSequences.size();i++) {
-            quest.getChild().add(indexOfModification, refJava.getChild().get(refSequences.size()-1-i));
+            questionnaire.getChild().add(indexOfModification, referencedQuestionnaire.getChild().get(refSequences.size()-1-i));
         }
-        logger.info("Sequences from {} inserted", ref);
+        logger.info("Sequences from {} inserted", reference);
+
+        // Add variables
+        List<VariableType> refVariables = referencedQuestionnaire.getVariables().getVariable();
+        refVariables.forEach(variable -> questionnaire.getVariables().getVariable().add(variable));
+        logger.info("Variables from {} inserted", reference);
 
-        // 2 - Add variables
-        List<VariableType> refVariables = refJava.getVariables().getVariable();
-        refVariables.stream().forEach(var->quest.getVariables().getVariable().add(var));
-        logger.info("Var from {} inserted", ref);
+        // Add code lists
+        List<CodeList> refCodesList = referencedQuestionnaire.getCodeLists().getCodeList();
+        refCodesList.forEach(codeList -> questionnaire.getCodeLists().getCodeList().add(codeList));
+        logger.info("CodeList from {} inserted", reference);
 
-        // 3 - Add codeslist
-        List<CodeList> refCodesList = refJava.getCodeLists().getCodeList();
-        refCodesList.stream().forEach(codeList->quest.getCodeLists().getCodeList().add(codeList));
-        logger.info("CodeList from {} inserted", ref);
+        // Filters defined on referenced questionnaire
+        questionnaire.getFlowControl().forEach(flowControlType -> {
+            // The 'IfTrue' property defines begin/end member references (separated with '-') of the filter
+            String[] flowControlBounds = flowControlType.getIfTrue().split("-");
+            if (flowControlBounds.length != 2) {
+                logger.error(
+                        "'IfTrue' value '{}' is not compliant with Pogues-Model specification in FlowControl '{}'",
+                        flowControlType.getIfTrue(), flowControlType.getId());
+            }
+            // Replace questionnaire reference by its first/last sequence
+            String beginMember = flowControlBounds[0];
+            String endMember = flowControlBounds[1];
+            if (beginMember.equals(reference)) {
+                beginMember = referencedQuestionnaire.getChild().get(0).getId();
```

Comment:
Not null check ?

Reply:
Good point (we should not trust the front too much). Improved: https://github.com/InseeFr/Pogues-Back-Office/pull/144/commits/42430636bcf4fb62c0928f6415ae9cd3d9f25b62